streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -382,7 +394,13 @@
                     40 * 1000,
                     atLeast(0),
                     ConfigDef.Importance.MEDIUM,
-                    REQUEST_TIMEOUT_MS_DOC);
+                    REQUEST_TIMEOUT_MS_DOC)
+            .define(PROCESSING_GUARANTEE_CONFIG,
+                    ConfigDef.Type.STRING,
+                    "at_least_once",
+                    in("at_least_once", "exactly_once"),
```

Comment:
That's a bad pattern in `ConsumerConfig` that we should not be trying to emulate. We also have the following which doesn't use magic values:

```
in(Sensor.RecordingLevel.INFO.toString(), Sensor.RecordingLevel.DEBUG.toString()),
```

Is the builder pattern switch happening in this release?